icu4c/source/i18n/smpdtfmt.cpp

```diff
@@ -721,19 +721,29 @@ void SimpleDateFormat::construct(EStyle timeStyle,
 
     UnicodeString timePattern;
     if (timeStyle >= kFull && timeStyle <= kShort) {
+        bool hasRgOrHcSubtag = false;
+        // also use DTPG if the locale has the "rg" or "hc" ("hours") subtag-- even if the overriding region
+        // or hour cycle is the same as the one we get by default, we go through the DateTimePatternGenerator
+        char dummy;
+        UErrorCode dummyErr1 = U_ZERO_ERROR, dummyErr2 = U_ZERO_ERROR;
+        if (locale.getKeywordValue("rg", &dummy, 0, dummyErr1) > 0 || locale.getKeywordValue("hours", &dummy, 0, dummyErr2) > 0) {
```

Comment:
In ICU4C, don't we need to check for both "hours" and "hc"? Per [Locale::getKeywordValue() docs](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1Locale.html#a6a504ee55c8f74fff0813921ae2e0330), "ICU4C doesn't do automatic conversion between legacy and Unicode keywords and values in getters and setters (as opposed to ICU4J)"

Reply:
But I guess you have test cases using "hc", so maybe not...

Reply:
> In ICU4C, don't we need to check for both "hours" and "hc"? Per [Locale::getKeywordValue() docs](https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classicu_1_1Locale.html#a6a504ee55c8f74fff0813921ae2e0330),

Locale::getKeywordValue("hours")
is the same as
Locale::getUnicodeKeywordValue("hc")

The doc means Locale::getKeywordValue("hc") will not automatically convert to
Locale::getKeywordValue("hours") and Locale::getKeywordValue("hc") does not work.

Reply:
Ah OK, thanks

Reply:
As you can see, I have unit tests that do it both ways.  Both `hc` and `hours` work.